File: eCar/eCar.Services/Database/Review.cs
using System;
using System.Collections.Generic;

namespace eCar.Services.Database;

public partial class Review
{
    public int Id { get; set; }

    public int? Value { get; set; }

    public string? Description { get; set; }

    public int ReviewsId { get; set; }

    public int ReviewedId { get; set; }

    public int RouteId { get; set; }

    public virtual Driver Reviewed { get; set; } = null!;

    public virtual Client Reviews { get; set; } = null!;

    public virtual Route Route { get; set; } = null!;
}

File: eCar/UI/ecar_mobile/lib/models/Review/review.dart
import 'package:ecar_mobile/models/Client/client.dart';
import 'package:ecar_mobile/models/Driver/driver.dart';
import 'package:ecar_mobile/models/Route/route.dart';
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  int? id;
  int? value;
  String? description;
  int? reviewsId;
  int? reviewedId;
  DateTime? addedDate;
  int? routeId;
  Driver? reviewed;
  Client? reviews;
  Route? route;

  Review(
      {this.id,
      this.value,
      this.description,
      this.reviewsId,
      this.reviewedId,
      this.addedDate,
      this.routeId,
      this.reviewed,
      this.reviews,
      this.route});

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);

  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
